big-red-shuttle/StopsViewController.swift

```diff
@@ -263,38 +264,52 @@ class StopsViewController: UIViewController, UITableViewDelegate, UITableViewDat
             dismissPopUpView(newPopupStop: newStop, fullyDismissed: selectedStop == newStop)
         }
 
+        if let markerCurr = selectedMarker {
+            let iconView = markerCurr.iconView as! IconView
+            markerUnselectedAnimation(iconView: iconView)
+        }
+
         let iconView = marker.iconView as! IconView
         if !iconView.clicked! {
-            UIButton.animate(withDuration: 0.15, animations: {
-                iconView.circleView.transform = CGAffineTransform(scaleX: 1.1, y: 1.1)
-                CATransaction.begin()
-                CATransaction.setDisableActions(true)
-                iconView.smallGrayCircle.strokeColor = UIColor.brsred.cgColor
-                CATransaction.commit()
-            })
-            { (finished:Bool) in
-                if finished {
-                    iconView.clicked = true
-                }
-            }
+            markerSelectedAnimation(iconView: iconView)
+            selectedMarker = marker
         } else {
-            UIButton.animate(withDuration: 0.15, animations: {
-                iconView.circleView.transform = CGAffineTransform(scaleX: 1.0, y: 1.0)
-                CATransaction.begin()
-                CATransaction.setDisableActions(true)
-                iconView.smallGrayCircle.strokeColor = UIColor.iconlightgray.cgColor
-                CATransaction.commit()
-            })
-            { (finished:Bool) in
-                if finished {
-                    iconView.clicked = false
-                }
-            }
+            markerUnselectedAnimation(iconView: iconView)
+            selectedMarker = nil
         }
-        
         return true
     }
     
+    func markerSelectedAnimation(iconView:IconView) {
+        UIButton.animate(withDuration: 0.15, animations: {
+            iconView.circleView.transform = CGAffineTransform(scaleX: 1.1, y: 1.1)
+            CATransaction.begin()
+            CATransaction.setDisableActions(true)
+            iconView.smallGrayCircle.strokeColor = UIColor.brsred.cgColor
+            CATransaction.commit()
+        })
+        { (finished:Bool) in
+            if finished {
+                iconView.clicked = true
+            }
+        }
+    }
```

Comment:
Squash content from `markerSelectedAnimation` and `markerUnselectedAnimation` into one helper function